client.go

```diff
@@ -0,0 +1,204 @@
+package gev
+
+import (
+	"errors"
+	"github.com/Allenxuxu/gev/connection"
+	"github.com/Allenxuxu/gev/eventloop"
+	"github.com/Allenxuxu/gev/log"
+	"github.com/Allenxuxu/toolkit/sync"
+	"github.com/Allenxuxu/toolkit/sync/atomic"
+	"github.com/RussellLuo/timingwheel"
+	"github.com/libp2p/go-reuseport"
+	"golang.org/x/sys/unix"
+	"net"
+	"runtime"
+	"syscall"
+	"time"
+)
+
+type Connector struct {
+	workLoops   []*eventloop.EventLoop
+	opts        *Options
+	timingWheel *timingwheel.TimingWheel
+	running     atomic.Bool
+}
+
+func connect(network, address string) (int, error) {
+	addr, err := reuseport.ResolveAddr(network, address)
+	if err != nil {
+		return 0, err
+	}
+
+	var sa unix.Sockaddr
+
+	// net dial.go
+	var domain, typ int
+	switch ra := addr.(type) {
+	case *net.TCPAddr:
+		domain = unix.AF_INET
+		typ = unix.SOCK_STREAM
+		ipaddr := ra.IP.To4()
+		if len(ipaddr) == net.IPv4len {
+			addr := &unix.SockaddrInet4{Port: ra.Port}
+			copy(addr.Addr[:], ipaddr)
+			sa = addr
+		} else if len(ipaddr) == net.IPv6len {
+			addr := &unix.SockaddrInet6{Port: ra.Port}
+			copy(addr.Addr[:], ipaddr)
+			sa = addr
+		}
+	case *net.UnixAddr:
+		domain = unix.AF_UNIX
+		typ = unix.SOCK_STREAM
+		sa = &unix.SockaddrUnix{Name: ra.Name}
+
+	default:
+		return 0, errors.New("unsupported network/address type")
+	}
+
+	fd, err := unix.Socket(domain, typ, unix.PROT_NONE)
+	if err != nil {
+		log.Error("unix-socket err:", err)
+		return 0, err
+	}
+
+	if err = unix.SetNonblock(fd, true); err != nil {
+		log.Error("unix-setnonblock err:", err)
+		_ = unix.Close(fd)
+		return 0, err
+	}
+
+	err = unix.Connect(fd, sa)
+	if err != nil && err != unix.EINPROGRESS {
+		_ = unix.Close(fd)
+		return 0, err
+	} else if err != nil {
+		err = nil
+	}
+
+	l := time.After(time.Second * 5)
+	check := func() error {
+		var n int
+		for {
+			select {
+			case <-l:
+				err = errors.New("timeout")
+				_ = unix.Close(fd)
+				return err
+
+			default:
+				wFdSet := &unix.FdSet{}
+				wFdSet.Set(fd)
+
+				n, err = unix.Select(fd+1, wFdSet, wFdSet, nil, &unix.Timeval{Sec: 1, Usec: 0})
```

Comment:
use poller?

Reply:
i dont think ur poller is ready for that, select got easy timeout